lib/Migration/CacheUserDisplayNames.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2021 Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Migration;
+
+use OCA\Talk\Model\Attendee;
+use OCP\IDBConnection;
+use OCP\IUser;
+use OCP\IUserManager;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+
+class CacheUserDisplayNames implements IRepairStep {
+
+	/** @var IDBConnection */
+	protected $connection;
+	/** @var IUserManager */
+	protected $userManager;
+
+	public function __construct(IDBConnection $connection,
+								IUserManager $userManager) {
+		$this->connection = $connection;
+		$this->userManager = $userManager;
+	}
+
+	public function getName(): string {
+		return 'Cache the user display names';
+	}
+
+	public function run(IOutput $output): void {
+		$update = $this->connection->getQueryBuilder();
+		$update->update('talk_attendees')
+			->set('display_name', $update->createParameter('displayName'))
+			->where($update->expr()->eq('actor_type', $update->createParameter('actorType')))
+			->andWhere($update->expr()->eq('actor_id', $update->createParameter('actorId')));
+
+		$query = $this->connection->getQueryBuilder();
+		$query->select('actor_id')
+			->from('talk_attendees')
+			->where($query->expr()->eq('actor_type', $query->createNamedParameter(Attendee::ACTOR_USERS)))
+			->andWhere($query->expr()->eq('display_name', $query->createNamedParameter('')))
+			->groupBy('actor_id');
+
+		$result = $query->execute();
+		while ($row = $result->fetch()) {
+			$user = $this->userManager->get($row['actor_id']);
```

Comment:
not sure if we should worry about millions of users and make that an occ command instead of running during update

I guess it is less likely that we'll traverse all possible users (like LDAP) as it would mean every one exists at least once in a conversation

Reply:
Since this is only executed once after the update, I don't mind timeouts (as big instances need to update via occ anyway) and so this should be fine.

Reply:
Also in case it would ever timeout, a follow up run would continue from where it broke because of the `>andWhere($query->expr()->eq('display_name', $query->createNamedParameter('')))` part checking for empty names

Reply:
Yeah, makes sense.

My comment was more in the lines of "is the admin aware that this update will take longer than usual" as admins need to plan time for downtime. But we could add this to the release notes to make them aware.

Reply:
Well it's a simple query per users. We and other apps have done worse things. e.g. the 10.1 release copied all entries for participants to the attendees table and stuff like that.